### DISCUSSION TITLE
jinja2 non device/site specific VLAN?

### DISCUSSION THREAD

**Discussion by mostdaysarebaddays** [netbox-community/netbox#15564] [Help Wanted!]
Can I get just the full VLANs list, not those specific to VLAN site etc? Looking to just call 'ipam.vlans' but that gives an error.

I know you can use 'device.site.vlans.all()' for site specific but the VLAN in question 'generic' and in a VLAN group, not in any site.

(my goal here is to get the prefix assigned to a certain 'generic' VLAN for that site but need to get the VLAN first)

**Comment by candlerb** [comment]
> Looking to just call 'ipam.vlans' but that gives an error

Show the call, and show the error. It works for me.

```
# curl -sS -H "Authorization: Token $TOKEN" "$NETBOX/api/ipam/vlans/" | jq .
```

I am guessing from the context of your question that the problem may be with your use of the `pynetbox` REST client library. But that works for me too:

```
>>> for v in nb.ipam.vlans.all():
...     print(v)
```

**Reply by mostdaysarebaddays** [reply]
It works fine with pynetbox or api, I'm trying to call using 'render config' and jinja2 which only seems to be aware of its current device and not the greater Netbox.

**Reply by candlerb** [reply]
Ah right. In that case what you want was added in Netbox [v3.6.0](https://github.com/netbox-community/netbox/releases/v3.6.0), but you didn't say what version you're using.

> ## Enhancements
> ...
> #12814 - Expose NetBox models within ConfigTemplate rendering context

Documentation [here](https://docs.netbox.dev/en/stable/features/configuration-rendering/#context-data). It looks like `ipam.VLAN.objects.all()` is what you want.

**Reply by mostdaysarebaddays** [reply]
Thank you! I’m on 3.6 and I totally missed this. Hugely appreciated.
